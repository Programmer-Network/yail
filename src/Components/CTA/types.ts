import { IIconProps } from "Components/Icon/types";

export interface ICTAProps {
  IconComponent?: IIconProps;
  ButtonIconComponent?: IIconProps;
  iconClassName?: string;
  title: string;
  subtitle: string;
  buttonText?: string;
  onClick?: () => void;
}
