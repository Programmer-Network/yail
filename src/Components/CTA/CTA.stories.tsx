import { action } from "@storybook/addon-actions";

import CTA from ".";
import { ICTAProps } from "./types";

export default {
  title: "Components/CTA",
  parameters: {
    layout: "centered"
  },
  component: CTA
};

export const Default = (args: ICTAProps) => <CTA {...args} />;

Default.args = {
  title: "You haven't added any experience",
  subtitle:
    "Adding experience will expose your profile to recruiters and like-minded developers looking to team up for projects",
  buttonText: "Get Started",
  onClick: action("onClick"),
  IconComponent: {
    iconName: "IconAddCircle",
    dataTestId: "icon-add-circle"
  },
  ButtonIconComponent: {
    iconName: "IconAddCircle",
    dataTestId: "icon-add-circle"
  }
};
