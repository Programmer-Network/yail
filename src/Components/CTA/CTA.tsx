import classNames from "classnames";
import { FC } from "react";

import { Button } from "../Button";
import { Icon } from "../Icon";
import { H3, Paragraph } from "../Typography";
import { ICTAProps } from "./types";

const CTA: FC<ICTAProps> = (
  {
    ButtonIconComponent,
    IconComponent,
    iconClassName,
    title,
    subtitle,
    buttonText,
    onClick
  } = {
    iconClassName: "",
    IconComponent: undefined,
    ButtonIconComponent: undefined,
    title: "",
    subtitle: "",
    buttonText: "",
    onClick: () => null
  }
) => {
  return (
    <div className='flex items-center justify-center'>
      <div className='flex flex-col items-center text-center'>
        <div className='flex flex-col items-center justify-center gap-4'>
          <div className='flex flex-col gap-2'>
            {IconComponent && (
              <Icon
                iconName={IconComponent.iconName}
                dataTestId={IconComponent.dataTestId}
                className={classNames(
                  "text-primary mx-auto w-14 md:w-28",
                  iconClassName
                )}
              />
            )}

            <H3>{title}</H3>
            <Paragraph className='text-secondary mx-auto max-w-md text-center'>
              {subtitle}
            </Paragraph>
          </div>

          {buttonText && onClick && (
            <Button onClick={onClick}>
              <span className='flex items-center gap-2'>
                {ButtonIconComponent && (
                  <Icon
                    iconName={ButtonIconComponent.iconName}
                    className='w-4'
                    aria-hidden='true'
                  />
                )}
                {buttonText}
              </span>
            </Button>
          )}
        </div>
      </div>
    </div>
  );
};

export { CTA };
