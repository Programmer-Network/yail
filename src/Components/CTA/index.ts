export { default } from "./CTA";
export * from "./types";
