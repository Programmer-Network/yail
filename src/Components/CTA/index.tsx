import classNames from "classnames";
import { FC } from "react";

import Button from "Components/Button";
import { H3, Paragraph } from "Components/Typography";

import { ICTAProps } from "./types";

const CTA: FC<ICTAProps> = (
  { Icon, ButtonIcon, iconClassName, title, subtitle, buttonText, onClick } = {
    iconClassName: "",
    Icon: undefined,
    ButtonIcon: undefined,
    title: "",
    subtitle: "",
    buttonText: "",
    onClick: () => null
  }
) => {
  return (
    <div className='yl-flex yl-items-center yl-justify-center'>
      <div className='yl-flex yl-flex-col yl-items-center yl-text-center'>
        <div className='yl-flex yl-flex-col yl-items-center yl-justify-center yl-gap-4'>
          <div className='yl-flex yl-flex-col yl-gap-2'>
            {Icon && (
              <Icon
                className={classNames(
                  "yl-mx-auto yl-w-14 yl-text-primary md:yl-w-28",
                  iconClassName
                )}
              />
            )}

            <H3>{title}</H3>
            <Paragraph className='yl-mx-auto yl-max-w-md yl-text-secondary yl-text-center'>
              {subtitle}
            </Paragraph>
          </div>

          {buttonText && onClick && (
            <Button onClick={onClick}>
              <span className='yl-flex yl-items-center yl-gap-2'>
                {ButtonIcon && (
                  <ButtonIcon className='yl-w-4' aria-hidden='true' />
                )}
                {buttonText}
              </span>
            </Button>
          )}
        </div>
      </div>
    </div>
  );
};

export default CTA;
