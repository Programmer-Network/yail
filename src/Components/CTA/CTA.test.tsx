import { fireEvent, render, screen } from "@testing-library/react";
import { vi } from "vitest";

import CTA from "./";

describe("CTA component", () => {
  const title = "Test Title";
  const subtitle = "Test Subtitle";

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(<CTA title={title} subtitle={subtitle} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders title and subtitle", () => {
    render(<CTA title={title} subtitle={subtitle} />);
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(subtitle)).toBeInTheDocument();
  });

  test("conditionally renders icons", () => {
    render(
      <CTA
        title={title}
        subtitle={subtitle}
        IconComponent={{
          iconName: "IconSpinner",
          dataTestId: "icon-spinner"
        }}
        ButtonIconComponent={{
          iconName: "IconSpinner",
          dataTestId: "icon-spinner"
        }}
      />
    );

    expect(screen.getByTestId("icon-spinner")).toBeInTheDocument();
  });

  test("displays button with text and handles click", () => {
    const buttonText = "Click Me";
    const handleClick = vi.fn();
    render(
      <CTA
        title={title}
        subtitle={subtitle}
        buttonText={buttonText}
        onClick={handleClick}
      />
    );
    const button = screen.getByText(buttonText);
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(2);
  });
});
