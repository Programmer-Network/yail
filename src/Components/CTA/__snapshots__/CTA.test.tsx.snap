// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTA component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="yl:flex yl:items-center yl:justify-center"
  >
    <div
      class="yl:flex yl:flex-col yl:items-center yl:text-center"
    >
      <div
        class="yl:flex yl:flex-col yl:items-center yl:justify-center yl:gap-4"
      >
        <div
          class="yl:flex yl:flex-col yl:gap-2"
        >
          <h3
            class="yl:text-xl yl:sm:text-2xl yl:md:text-3xl yl:text-text yl:break-words yl:overflow-hidden yl:font-bold"
          >
            Test Title
          </h3>
          <p
            class="yl:text-secondary yl:break-words yl:overflow-hidden yl:mx-auto yl:max-w-md yl:text-secondary yl:text-center"
          >
            Test Subtitle
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
